conda/common/path/__init__.py

```diff
@@ -406,12 +367,136 @@ def is_package_file(path):
     return path[-6:] == ".conda" or path[-8:] == ".tar.bz2"
 
 
-def path_identity(
-    paths: PathType | Iterable[PathType] | None,
-) -> str | tuple[str, ...] | None:
+def path_identity(paths: PathType | PathsType | None) -> str | tuple[str, ...] | None:
     if paths is None:
         return None
     elif isinstance(paths, (str, os.PathLike)):
         return os.path.normpath(paths)
     else:
         return tuple(os.path.normpath(path) for path in paths)
+
+
+def _path_to(
+    paths: PathType | PathsType | None,
+    prefix: PathType | None,
+    *,
+    cygdrive: bool,
+    to_unix: bool,
+) -> str | tuple[str, ...] | None:
+    if paths is None:
+        return None
+
+    # short-circuit if we don't get any paths
+    paths = paths if isinstance(paths, (str, os.PathLike)) else tuple(paths)
+    if not paths:
+        return "." if isinstance(paths, (str, os.PathLike)) else ()
+
+    if on_win and prefix is None:
+        from ...base.context import context
+
+        prefix = context.target_prefix
+
+    if to_unix:
+        from_pathsep = ntpath.pathsep
+        cygpath_arg = "--unix"
+        cygpath_fallback = nt_to_posix
+        to_pathsep = posixpath.pathsep
+        to_sep = posixpath.sep
+    else:
+        from_pathsep = posixpath.pathsep
+        cygpath_arg = "--windows"
+        cygpath_fallback = posix_to_nt
+        to_pathsep = ntpath.pathsep
+        to_sep = ntpath.sep
+
+    # It is very easy to end up with a bash in one place and a cygpath in another due to e.g.
+    # using upstream MSYS2 bash, but with a conda env that does not have bash but does have
+    # cygpath.  When this happens, we have two different virtual POSIX machines, rooted at
+    # different points in the Windows filesystem.  We do our path conversions with one and
+    # expect the results to work with the other.  It does not.
+
+    # TODO: search prefix for cygpath instead of deriving it from bash
+    bash = which("bash")
+    cygpath = str(Path(bash).parent / "cygpath") if bash else "cygpath"
+    joined = (
+        str(paths)
+        if isinstance(paths, (str, os.PathLike))
+        else from_pathsep.join(map(str, paths))
+    )
+
+    try:
+        # if present, use cygpath to convert paths since its more reliable
+        converted = subprocess.run(
```

Comment:
Is there a reason we aren't using  `conda.gateways.subprocess.subprocess_call` here?

Reply:
Counter argument, why should we use `conda.gateways.subprocess.subprocess_call` and not move to deprecate it? It is not at all clear that it does anything better or safer. It appears to have initially been implemented to do a better job handling signals, see https://github.com/conda/conda/pull/4447 and https://stackoverflow.com/questions/13593223/making-sure-a-python-script-with-subprocesses-dies-on-sigint, at this point it smells like old Python 2 code.

Reply:
Fair point!

Maybe we should at least update the comment for `conda.gateways.subprocess.subprocess_call` then:

> This utility function should be preferred for all conda subprocessing.

This can be saved for another pull request though.